indico/modules/files/controllers.py

```diff
@@ -55,7 +62,7 @@ def get_file_context(self):
         return ``['event', EVENTID]`` so the file gets stored under
         ``event/EVENTID/...`` in the storage backend.
         """
-        raise NotImplementedError
+        return 'event', self.event.id, 'regform', self.regform.id, 'registration'
```

Comment:
This particular mixin is quite generic and not only for registrations. You need to create a new mixin (in the registration module):

```py
class UploadRegistrationFileMixin(UploadFileMixin):
    @use_kwargs({
        'token': UUIDString(load_default=None),
    }, location='query')
    def _process_args(self, token):
        self.existing_registration = self.regform.get_registration(uuid=token) if token else None

    def get_file_context(self):
        return 'event', self.event.id, 'regform', self.regform.id, 'registration'
```

Then you can simply use that in the RHs.